### PR TITLE
nixos/fonts: rename fonts.fonts option to fonts.packages, other cleanups 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

fonts.fonts is not a great name and this also resolves a TODO.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by asymmetric — CHANGES_REQUESTED** [review]
Seems like you forgot some (ignoring the last one, which is a release note):

```
❯ rg "fonts\.fonts"
pkgs/tools/typesetting/tex/texlive/combine.nix
106:    # useful for inclusion in the `fonts.fonts` nixos option or for use in devshells

nixos/modules/services/x11/desktop-managers/pantheon.nix
309:      fonts.fonts = [

nixos/doc/manual/release-notes/rl-2205.section.md
291:- `fonts.fonts` no longer includes ancient bitmap fonts when both `config.services.xserver.enable` and `config.nixpkgs.config.allowUnfree` are enabled.
296:    fonts.fonts = [
```

Also 
https://github.com/NixOS/nixpkgs/tree/87cf8b016d16e2c6de851fd1e52676c0eea54c7b/nixos/modules/services/ttys/kmscon.nix#L114

Also, should we handle this gracefully, with a deprecation message?

And it should be mentioned in the release notes for the next NixOS version, I believe?

**Comment by SuperSandro2000** [comment]
I am not sure why the evaluation succeeded, it shouldn't have with those missed.

> Also, should we handle this gracefully, with a deprecation message?

This should already be the case with the mkRenamedOptionModule

> And it should be mentioned in the release notes for the next NixOS version, I believe?

Upsi, missed that. I thought I already wrote that, maybe I didn't commit it.

File: nixos/modules/config/fonts/packages.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  options = {
+    fonts = {
+      packages = lib.mkOption {
+        type = with lib.types; listOf path;
+        default = [];
+        example = lib.literalExpression "[ pkgs.dejavu_fonts ]";
+        description = lib.mdDoc "List of primary font packages.";
+      };
+
+      enableDefaultFonts = lib.mkOption {
```
Comment (asymmetric) [inline nixos/modules/config/fonts/packages.nix:21]: Should this then become `enableDefaultPackages`? Or actually, I think `enableDefault` is even better.

**Review by asymmetric — COMMENTED** [review]
One final question.

File: nixos/modules/config/fonts/packages.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/config/fonts/packages.nix:21]: I thought about this and didn't think of enableDefaultPackages, enableDefault was a little bit to ambigious for me

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 2]
```diff
@@ -121,6 +121,8 @@
 
 - `buildGoModule` `go-modules` attrs have been renamed to `goModules`.
 
+- The `fonts.fonts` option has been renamed to `fonts.packages`.
```
Comment (asymmetric) [inline nixos/doc/manual/release-notes/rl-2311.section.md:124]: You should mention `enableDefaultPackages` too, no?

**Review by asymmetric — CHANGES_REQUESTED** [review]

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 2 above)
Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2311.section.md:124]: derp

**Review by asymmetric — APPROVED** [review]
LGTM!

**Comment by xaverdh** [comment]
It would be good to get an error that explains what the options were renamed to. When rebuilding my config I just got
``error: The option `fonts.enableDefaultFonts' does not exist.``
and had to look at the source code to find the new name

**Comment by asymmetric** [comment]
Isn't this supposed to be handled by [`mkRenamedOptionModule`](https://github.com/NixOS/nixpkgs/pull/244332/files#diff-a4dd5830f8140e1bf124d34c3f83538aac66edf96bf0eb373eae8a76fc42bf5bR9)?

**Comment by xaverdh** [comment]
> Isn't this supposed to be handled by [`mkRenamedOptionModule`](https://github.com/NixOS/nixpkgs/pull/244332/files#diff-a4dd5830f8140e1bf124d34c3f83538aac66edf96bf0eb373eae8a76fc42bf5bR9)?

Ideed, quite odd that it didn't work!

**Comment by xaverdh** [comment]
It worked for `fonts.fonts` by the way. Not sure what is going on there

File: nixos/modules/config/fonts/packages.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.fonts;
+in
+{
+  imports = [
+    (lib.mkRemovedOptionModule [ "fonts" "enableCoreFonts" ] "Use fonts.packages = [ pkgs.corefonts ]; instead.")
+    (lib.mkRenamedOptionModule [ "fonts" "enableDefaultFonts " ] [ "fonts" "enableDefaultPackages" ])
```
Comment (SuperSandro2000) [inline nixos/modules/config/fonts/packages.nix:9]: There is an extra space here :(

**Comment by SuperSandro2000** [comment]
#245380

File: nixos/modules/config/fonts/packages.nix (hunk 3 above)
Comment (asymmetric) [reply nixos/modules/config/fonts/packages.nix:9]: :sob:

Comment (linuxmobile) [reply nixos/modules/config/fonts/packages.nix:9]: Waiting the fix to enableDefaultPackages :3

Comment (asymmetric) [reply nixos/modules/config/fonts/packages.nix:9]: @linuxmobile https://github.com/NixOS/nixpkgs/pull/245380 or do you mean something else?

**Pull request merged**